docs/source/topics/proc_deploying-sample-application-with-odo.adoc

```diff
@@ -0,0 +1,60 @@
+[id="deploying-sample-application-with-odo_{context}"]
+= Deploying a sample application with `odo`
+
+You can use OpenShift Do ([command]`odo`) to create OpenShift projects and applications from the command line.
+This procedure deploys a sample Java application to the OpenShift cluster running in the {prod} virtual machine.
+
+.Prerequisites
+
+* [command]`odo` is installed.
+For more information, see link:{odo-docs-url-installing}[Installing `odo`] in the [command]`odo` documentation.
+* The {prod} virtual machine is running.
+For more information, see <<starting-the-virtual-machine_{context}>>.
+* You must log in through [command]`oc` as the `developer` user.
+For more information, see <<accessing-the-openshift-cluster-with-oc_{context}>>.
+
+.Procedure
+
+To deploy a sample Java application through [command]`odo`, follow these steps:
```

Comment:
This is both redundant with https://docs.openshift.com/container-platform/4.3/cli_reference/openshift_developer_cli/creating-a-single-component-application-with-odo.html , and at the same time slightly different (java instead of nodejs, odo create --git rather than direct git clone call, ...) I don't know if we should duplicate all these steps in crc documentation ?

Reply:
This is also greatly simplified and doesn't go into detail.

The main idea here is to quickly demonstrate how to create and deploy a sample application using both together. Without any documentation "gluing" these concepts, the idea remains somewhat abstract.

If we only want to document how to connect `odo` and `crc`, then there's really almost nothing to document -- start CRC, then `odo login -u developer -p developer` -- that's it. I thought we should go into a little bit more detail here to have the user end up with a visible deployment.

Reply:
Ok, you're our doc owner, so that's your call, fine with me :)

Reply:
I do agree that it's essentially redundant, but that's not always a bad thing. :)

One of the ideas here is to use minimal steps to get something visibly working. Fortunately, all of the steps described are stable and there shouldn't be any maintenance cost re: keeping it up to date (unless we keep having memory constraint issues, as mentioned by @praveenkumar).

Reply:
@cfergeau odo also has a guide with Java https://docs.openshift.com/container-platform/4.3/cli_reference/openshift_developer_cli/creating-a-multicomponent-application-with-odo.html and a list of sample apps in different languages https://docs.openshift.com/container-platform/4.3/cli_reference/openshift_developer_cli/using-sample-applications.html

@robin-owen regarding `odo create --git` vs `git clone`, as it was mentioned in odo channel, `git clone` would probably be quicker on push for users. It adds more steps to the procedure so your call. :)

Reply:
@boczkowska Right, that's the issue I was trying to figure out myself. ;)

Considering this is more of a demonstration than a detailed guide, I considered less commands to type (read: steps to follow) was appropriate. More information is always available from the `odo` docs directly. :)